HttpJs handle http request with r.status == 0

AdrianV opened the pull request:
or `r.response == null`.
this will fix #8749 
Additional it will handle any case when `r.response == null`. The later would fix the case for a http request which could not be resolved too, But handling the particular case when `r.status == 0` will give the right error message.

AdrianV committed:
HttpJs handle http request with r.status == 0

or `r.response == null`.
this will fix #8749 
Additional it will handle any case when `r.response == null`. The later would fix the case for a http request which could not be resolved too, But handling the particular case when `r.status == 0` will give the right error message.

RealyUniqueName commented on std/haxe/http/HttpJs.hx:
```diff
@@ -74,7 +74,7 @@ class HttpJs extends haxe.http.HttpBase {
 				req = null;
 				success(Bytes.ofData(r.response));
-			} else if (s == null) {
+			} else if (s == null || s == 0) {
```
This should be changed to `|| (s == 0 && r.response == null)`
Or add `responseBytes = r.response != null ? Bytes.ofData(r.response) : null;` to the body of this `if`
That is needed to not lose response body in case browsers set `s = 0` in other weird edge cases (like local requests).

AdrianV replied:
I would prefer the first, because this is the case for the error message - you can't get a response if the browser is unable to resolve the url. Then the other weird cases would be covered in the `default` branch with a less explicit error message.

AdrianV committed:
rev HttpJs handle http request with r.status == 0 

or `r.response == null`.
this will fix HaxeFoundation#8749 
Additional it will handle any case when `r.response == null`. The later would fix the case for a http request which could not be resolved too, But handling the particular case when `r.status == 0` will give the right error message.
